src/backend/resources.rs

```diff
@@ -0,0 +1,270 @@
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d2" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d2"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d3" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d3"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ if name == "seq_mem_d4" => {
+                    *count_map
+                        .entry((
+                            Id::from("seq_mem_d4"),
+                            (**param_binding).clone(),
+                        ))
+                        .or_insert(0) += 1
+                }
+                _ => (),
+            },
+            CellType::Component { name: _name } => todo!(),
+            _ => (),
+        }
+    }
+    count_map
+}
+
+/// Writes a CSV to stdout with primitive count information
+/// generated by `gen_count_map`.
+/// TODO (priya): fix the output format and have it output to an actual file!
+fn write_csv(count_map: HashMap<(Id, Binding), u32>) {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+    let header = vec!["Primitive", "Count", "Parameters"];
+    wtr.write_record(header).ok();
+    let mut result: Vec<u8> = vec![];
+    for ((name, params), count) in count_map {
+        result.append(&mut name.id.to_string().as_str().as_bytes().to_vec());
+        result.append(&mut (", ").as_bytes().to_vec());
+        result.append(&mut count.to_string().as_str().as_bytes().to_vec());
+        result.append(&mut (", ").as_bytes().to_vec());
+        for (id, val) in params {
+            result.append(&mut id.id.to_string().into_bytes());
+            result.append(&mut (", ").as_bytes().to_vec());
+            result.append(&mut val.to_string().into_bytes());
+            result.append(&mut (", ").as_bytes().to_vec());
+        }
+        result.append(&mut ("\n").as_bytes().to_vec());
+        wtr.write_record([&result]).ok();
+        result = vec![];
+    }
+    wtr.flush().ok();
+}
+
+// Prints the estimated size (in bits) of the generated hardware along with a breakdown
+// of which primitives contributed to the total number.
+// TODO (priya): Add other primitives (need to discuss what reasonable size estimates are for them)
+fn estimated_size(count_map: HashMap<(Id, Binding), u32>) {
+    let mut estimated_size: u64 = 0;
+    let out = &mut (Box::new(io::stdout()) as Box<dyn Write>);
+    output("Summary of primitives:".to_string(), out);
+    for ((name, params), count) in count_map {
+        match name {
```

Comment:
You can just use `if` instead of `match ... if` arms since there is no benefit in doing the latter AFAIK

Reply:
Seconded! See my earlier comment for more of an explanation on why this is.

Reply:
I should also add if you wanted to this via the match statement instead of an if block you can use the `AsRef<&str>` implementation on `Id` to allow a direct comparison in the match. I.e. `match name` becomes `match name.as_ref()` and `_ if name == "x" =>` becomes `"x" =>`